wolfcrypt/src/dsa.c

```diff
@@ -346,7 +348,7 @@ int wc_MakeDsaParameters(WC_RNG *rng, int modulus_size, DsaKey *dsa)
      */
     if (err == MP_OKAY) {
         if (loop_check_prime)
-            err = mp_add_d(tmp2, 2*loop_check_prime, tmp2);
+            err = mp_add_d(tmp2, (mp_digit)(2*loop_check_prime), tmp2);
```

Comment:
clang-tidy has an interesting carp about this, which I haven't seen before:
```
wolfssl/wolfcrypt/src/dsa.c:351:34: warning: either cast from 'int' to 'mp_digit' (aka 'unsigned long') is ineffective, or there is loss of precision before the conversion [bugprone-misplaced-widening-cast]
err = mp_add_d(tmp2, (mp_digit)(2*loop_check_prime), tmp2);
                                 ^
```

Reply:
fixed